CMakeLists.txt

```diff
@@ -1,17 +1,14 @@
 # Icinga 2 | (c) 2012 Icinga GmbH | GPLv2+
 
-cmake_minimum_required(VERSION 2.8.12)
+# The CMake policies starting from 3.0 to 3.17 shouldn't affect the Icinga 2 build process in any way,
+# so the maximum CMake required version is set to 3.17. Anything above that may work, but we didn't check
+# the policies introduced in CMake 3.18 and later.
+cmake_minimum_required(VERSION 3.8...3.17)
```

Comment:
> Writing 3.8 here isn't a burden, it just describes the state of the CMake config with this PR more accurately.

@julianbrost The burden is that comment one has to write/read/understand **in order to** reflect "the state of the CMake config with this PR more accurately".

Again, why to throw away v3.8 "at some point"? Why not right now while on it?

Reply:
> @julianbrost The burden is that comment one has to write/read/understand **in order to** reflect "the state of the CMake config with this PR more accurately".

That sounds like you just hate the `...` syntax and you wouldn't even be happy with `cmake_minimum_required(VERSION 3.16...3.17)`.

I think it makes sense to say "we verified the policies up to 3.17 and they work fine" so that if 3.17 is used, it doesn't have to emulate 3.17 behavior.

> Again, why to throw away v3.8 "at some point"? Why not right now while on it?

Because the answer to "why is 3.8 required?" is "3.8 added support for [`CXX_STANDARD=17`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html)" whereas the answer to "why is 3.16 required?" is "well, actually it isn't".